src/either.js

```diff
@@ -2,28 +2,43 @@ import {
   untag,
 } from './pair'
 
+import type {
+  Data,
+  Type,
+  TypeChecker,
+} from './type'
+
+import {
+  createType,
+} from './type'
+
 import {
   ap,
   cond,
+  createMatch,
   eq,
 } from './cond'
 
-type LeftT<A>  = ['Left',  A]
-type RightT<B> = ['Right', B]
-export type Either<A, B> = LeftT<A> | RightT<B>
+type LeftT<A>  = Type<'Left',  A>
+type RightT<A> = Type<'Right', A>
+export type EitherT<A, B> = LeftT<A> | RightT<B>
+
+const Left: Data<LeftT<*>, mixed> = createType('Left')
```

Comment:
Shouldn’t it be `any` instead of `*` in `LeftT<*>`? Flow should not tie the type to the first usage of `Left`